test/test_cost.py

```diff
@@ -233,6 +234,38 @@ def set_advantage_keys_through_loss_test(
             )
 
 
+@pytest.mark.parametrize("device", get_default_devices())
+@pytest.mark.parametrize("vmap_randomness", (None, "different", "same"))
+@pytest.mark.parametrize("dropout", (0.0, 0.1))
+def test_loss_vmap_random(device, vmap_randomness, dropout):
+    class VmapTestLoss(LossModule):
+        def __init__(self):
+            super().__init__()
+            layers = [nn.Linear(4, 4), nn.ReLU()]
+            if dropout > 0.0:
+                layers.append(nn.Dropout(dropout))
+            layers.append(nn.Linear(4, 4))
+            net = nn.Sequential(*layers).to(device)
+            model = TensorDictModule(net, in_keys=["obs"], out_keys=["action"])
+            self.convert_to_functional(model, "model", expand_dim=4)
+            self.vmap_model = _vmap_func(
+                self.model, (None, 0), randomness=self.vmap_randomness
+            )
+
+        def forward(self, td):
+            out = self.vmap_model(td, self.model_params)
+            return {"loss": out["action"].mean()}
+
+    loss_module = VmapTestLoss()
+    td = TensorDict({"obs": torch.randn(3, 4).to(device)}, [3])
+
+    # If user sets vmap randomness to a specific value
+    if vmap_randomness in ("different", "same") and dropout > 0.0:
+        loss_module.set_vmap_randomness(vmap_randomness)
+
+    loss_module(td)["loss"]
```

Comment:
maybe let's test that things actually fail if we don't call the `set_vmap_randomness` before?

Reply:
If we don't do `loss_module.set_vmap_randomness(vmap_randomness) `and have a Module that uses randomness vmap_randomness sets default to "different". So its only if the user wants a specific vmap_randomness. I think there is no case in which we should expect an error, only if the user sets vmap_randomness manually to "error" and uses dropout for example. 

I can add a test for that but not sure if thats what you meant.

Reply:
> only if the user sets vmap_randomness manually to "error" and uses dropout for example

Yes that is what I meant. Here we only test that the code runs, but we're not really checking that it would have been broken had we done things differently.

Reply:
added it!